drivers/CoreLED/source/CoreLED.cpp

```diff
@@ -0,0 +1,47 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "CoreLED.h"
+
+using namespace leka;
+
+auto CoreLED::setColor(RGB color) -> void
+{
+	_color = color;
+}
+
+auto CoreLED::isOn() const -> bool
+{
+	return _is_on;
+}
+
+auto CoreLED::showColor() -> void
+{
+	uint8_t byte_zero  = 0x00;
+	uint8_t brightness = 0x32;
+	auto start_frame   = std::to_array({byte_zero, byte_zero, byte_zero, byte_zero});
+	_spi.write(start_frame);
+
+	for (auto i = 0; i < (N_LEDS * 3); i += 3) {
+		auto data = std::to_array({brightness, _color.red, _color.green, _color.blue});
+		_spi.write(data);
+	}
+
+	auto reset_frame = std::to_array({byte_zero, byte_zero, byte_zero, byte_zero});
+	_spi.write(reset_frame);
+
+	// At least N_LEDS/2 bits of 0 for the end frame ()
+	auto end_frame = std::to_array({byte_zero, byte_zero});
```

Comment:
```suggestion
	constexpr auto end_frame = std::to_array({byte_zero, byte_zero});
```

je sortirai les start/reset et end frame pour les mettre en static constexpr en variable membre. pas besoin de les créer à chaque fois et ça rendra la lecture plus simple

Reply:
La fonction spi.write() n'accepte pas de constexpr, il dit " No viable conversion from 'const std::array<unsigned char, 4>' to 'std::span<uint8_t>' "
J'ai donc quand meme instancié en variable membre mais avec une déclaration classique.